BLSQ/iaso
# IA-2841 mapping select multiple to boolean

If you don't think about adding calculate you can't export the submissions containing a multi select. To remove the need about thinking about that "upfront", we will allow to map multi select to boolean type data element.

TODO :

- [x] add tests for the export
- [x] preserve mapping of "multiple choices" when uploading a new version
- implement this for other mappings
   - [ ] support multiple choices for event tracker (warn stages)
   - [ ] support multiple choices for aggregate (warn category combo option)

Also removed redux and used react query and hooks
No typescript for the moment.

Related JIRA tickets : IA-2841 IA-2952 IA-2953

## Self proofreading checklist


## Doc

## Changes

- udpate the seed command to show case that (xlsform with select_multiple, mapping for the boolean)
![image](https://github.com/BLSQ/iaso/assets/371692/2618709e-1d15-4d02-83c2-4e9258f6f5b8)

- update the mapping screen to display the coverage and options below.
![image](https://github.com/BLSQ/iaso/assets/371692/15e9120d-ef1f-4f67-80cd-1e70c5219f92)

- the backend for storing has remain the same (quite generic)
- update the exporter to handle that

## How to test

1. launch the seed command (not it won't update the xlsform on an existing seeded account)
   - `docker-compose run --rm iaso manage seed_test_data --mode=seed --dhis2version=2.39`
3. find the "Community Verification Satisfaction form"
4. find a submission
5. export it 
6. check the results in dhis2 
   - take the "export id" on the instance, it's the eventId in the url : https://play.dhis2.org/2.39.4.1/dhis-web-capture/index.html#/viewEvent?viewEventId=aMulsxS3imr
   - ![image](https://github.com/BLSQ/iaso/assets/371692/a210bb78-6a1c-4c59-ba7b-92ed84c74ab8)
   - ![image](https://github.com/BLSQ/iaso/assets/371692/7d4c0f8f-517d-49ef-ad70-eb90026e92f8)



## Print screen / video

Upload here print screens or videos showing the changes

## Notes

This won't work if the choice name contains a whitespace.
